panel/src/components/Views/LoginView.vue

```diff
@@ -24,21 +30,46 @@ export default {
 		methods: Array,
 		pending: Object
 	},
+	data() {
+		return {
+			issue: ""
+		};
+	},
 	computed: {
 		form() {
 			if (this.pending.email) {
 				return "code";
 			}
 
-			if (!this.$user) {
-				return "login";
+			return "login";
+		},
+		viewClass() {
```

Comment:
Is there a reason why we need these classes on `k-view` and cannot simply stick them on their own components?

Reply:
They might be necessary for custom login form components. E.g. if you need an entirely different layout that isn't centered. Also changing it now would be a breaking change.

Reply:
But those wouldn't have control over which classes to add anyways, it's always just those two from us. So not entering would and could be done via CSS anyways. It's a bit of a breaking change - but not more than many of the other changes in this PR. So I think that would be fine.

Reply:
Yes, the idea is: A user defines a custom login form component and then also needs CSS changes to the view. Without the class on the view, there's no way to style it.

Reply:
> It's a bit of a breaking change - but not more than many of the other changes in this PR

Are there any other breaking changes in the PR?

Reply:
Ok, then we should change this in 3.8 only.

Since people extend all kind of things in the Panel, almost all changes are kind of breaking changes - e.g. removing `this.issue` and corresponding functionality from the child components. I'm totally on your side that we need to make these and that the risk is low - but technically the Panel is very undefined in terms of what developers can rely on, what's off limits etc - and so almost any change becomes a breaking change in some way.

Reply:
I first also thought that removing `this.issue` from child components was a breaking change, but it isn't. If the child component displays its own `LoginAlert`, that's completely fine. Only the `LoginCode` component must not do that, but for that we don't have a plugin API.

Reply:
But that's what I mean, the Panel isn't really defined in terms of plugin APIs - we've encouraged users to extend and overwrite all kind of components. Hence, if someone extends LoginCode it might work unexpectedly different for them now.

Reply:
We do have a plugin API for Login, but none for LoginCode.
I'd still be fine to move the fix to 3.8.0.

Reply:
We do have `plugins.components` which we have encouraged a lot to do, so also `LoginCode` can be extended/replaced via that API.